tests/reader-errors.rs

```diff
@@ -325,3 +331,316 @@ mod syntax {
+                }
+
+                #[cfg(feature = "async-tokio")]
+                #[tokio::test]
+                async fn async_tokio() {
+                    let mut buf = Vec::new();
+                    let mut reader = Reader::from_str(concat!($xml, "<x/>"));
+                    reader.config_mut().enable_all_checks(true);
+                    reader
+                        .read_event_into_async(&mut buf)
+                        .await
+                        .expect("first .read_event_into_async()");
+                    match reader.read_event_into_async(&mut buf).await {
+                        Err(Error::IllFormed(cause)) => {
+                            assert_eq!(cause, $cause);
+                            assert_eq!(reader.buffer_position(), $pos);
+                        }
+                        x => panic!("Expected `Err(IllFormed(_))`, but got {:?}", x),
+                    }
+                    assert_eq!(
+                        reader.read_event_into_async(&mut buf).await.expect(
+                            "parsing should be possible to continue after `Error::IllFormed`"
+                        ),
+                        Event::Empty(BytesStart::new("x"))
+                    );
+
+                    let mut reader = NsReader::from_str(concat!($xml, "<x/>"));
+                    reader.config_mut().enable_all_checks(true);
+                    reader
+                        .read_resolved_event_into_async(&mut buf)
+                        .await
+                        .expect("first .read_resolved_event_into_async()");
+                    match reader.read_resolved_event_into_async(&mut buf).await {
+                        Err(Error::IllFormed(cause)) => {
+                            assert_eq!(cause, $cause);
+                            assert_eq!(reader.buffer_position(), $pos);
+                        }
+                        x => panic!("Expected `Err(IllFormed(_))`, but got {:?}", x),
+                    }
+                    assert_eq!(
+                        reader
+                            .read_resolved_event_into_async(&mut buf)
+                            .await
+                            .expect(
+                                "parsing should be possible to continue after `Error::IllFormed`"
+                            )
+                            .1,
+                        Event::Empty(BytesStart::new("x"))
+                    );
+                }
+            }
+        };
+    }
+
+    // IllFormedError::MissedVersion is generated lazily when you call `BytesDecl::version()`
+
+    err!(missed_doctype_name1("<!DOCTYPE>") => 9: IllFormedError::MissedDoctypeName);
+    //                                  ^= 9
+    err!(missed_doctype_name2("<!DOCTYPE \t\r\n>") => 13: IllFormedError::MissedDoctypeName);
+    //                                         ^= 13
+    ok!(missed_doctype_name3("<!DOCTYPE \t\r\nx>") => Event::DocType(BytesText::new("x")));
```

Comment:
Did you intend to leave these names as `missed_doctype_name3` (and below, `mismatched_end1`, and `double_hyphen_in_comment1`) for testing variants that are well-formed?

Reply:
Yes. The main motivation is that these names of the "test about `MissedDoctypeName` error number one... number two... and so on". The final test is a canary test that ensures that after all we able to parse correct representation without mentioned error.

I could wrap that tests into module named `missed_doctype_name` (snake cased error name), but I decide, that indent level would be not very comfort. The similar names of tests for the particular error is useful for running of only related tests via `cargo test ... -- missed_doctype_name`.

In the end, I thought that just a common name with a number is quite convenient and concise.